commands/delegate_submission.go

```diff
@@ -26,23 +26,23 @@ func checkDelegateSubmission(cmd *commandspb.DelegateSubmission) Errors {
 	return errs
 }
 
-func CheckUndelegateAtEpochEndSubmission(cmd *commandspb.UndelegateAtEpochEndSubmission) error {
-	return checkUndelegateAtEpochEndSubmission(cmd).ErrorOrNil()
+func CheckUndelegateSubmission(cmd *commandspb.UndelegateSubmission) error {
+	return checkUndelegateSubmission(cmd).ErrorOrNil()
 }
 
-func checkUndelegateAtEpochEndSubmission(cmd *commandspb.UndelegateAtEpochEndSubmission) Errors {
+func checkUndelegateSubmission(cmd *commandspb.UndelegateSubmission) Errors {
 	errs := NewErrors()
 
 	if cmd == nil {
-		return errs.FinalAddForProperty("undelegateAtEpochEnd_submission", ErrIsRequired)
+		return errs.FinalAddForProperty("undelegate_submission", ErrIsRequired)
 	}
 
-	if cmd.Amount <= 0 {
-		errs.AddForProperty("undelegateAtEpochEnd_submission.amount", ErrIsRequired)
+	if cmd.Method > commandspb.UndelegateSubmission_InAnger {
+		errs.AddForProperty("undelegate_submission.method", ErrIsRequired)
 	}
```

Comment:
Should you not still verify the cmd.Amount here?

Reply:
verify how? if it's zero, it's taken as undelegate all. beyond that what can it be?

Reply:
your absolutely right indeed.